File: 0001_0100/0070/0070.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s, n = f.read().split()
    n = int(n)
    l = len(s)
    t = ""
    if n > 0:
        while n and len(t) < 1023:
            n -= 1
            t += s
    else:
        n = -n
        t = s[: l // n] if l % n == 0 and s[: l // n] * n == s else "NO SOLUTION"
    q.write(t[:1023])

File: 0001_0100/0070/0070.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int
	var s string
	fmt.Fscan(f, &s, &n)
	r, l := "NO SOLUTION", len(s)
	if n > 0 {
		r = s
		for ; n > 1 && len(r) < 1023; n-- {
			r += s
		}
		if len(r) > 1023 {
			r = r[:1023]
		}
	} else if l%n == 0 && strings.Repeat(s[:-l/n], -n) == s {
		r = s[:-l/n]
	}
	fmt.Fprint(q, r)
}
